### DISCUSSION TITLE
Limit on array size?

### DISCUSSION THREAD

**Discussion by mlieshout** [spatie/ray#222] [Q&A] [answered]
Hi,

Does Ray have a limit on the array size?
I'm using ray v1.7.0

Great app btw!

<img width="1058" alt="Screenshot 2021-01-20 at 14 36 02" src="https://user-images.githubusercontent.com/994055/105182317-31acdb80-5b2d-11eb-8250-fd2cd32b4ee9.png">

**Comment by sinnbeck** [comment, marked as answer]
This is caused by VarCloner, which Ray uses to prepare the data. The default limit is 2500 items.
https://github.com/symfony/var-dumper/blob/5.x/Cloner/AbstractCloner.php#L189

Now if you take your arrays of 22 items and multiply with 114 (the cutoff) you get 2508, which is just beyond that limit.

It would be possible to extend this, but I am unsure if that is a good idea (that limit is most likely there for a reason)
